src/praisonai-agents/praisonaiagents/tools/nightmarket_tools.py

```
@@ -0,0 +1,131 @@
+        List of available API services with id, name, description, method, price, and seller info
+    """
+    try:
+        from urllib.request import urlopen, Request
+        from urllib.parse import urlencode
+
+        params = {"sort": sort}
+        if query:
+            params["search"] = query
+
+        url = f"{NIGHTMARKET_BASE_URL}/marketplace?{urlencode(params)}"
+        req = Request(url, headers={"Accept": "application/json"})
```

Ruff (0.15.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)